src/js/popup/popup.js

```diff
@@ -531,6 +530,86 @@ async function enableSettingsPanel() {
   }
 }
 
+function enableReportIssuePanel() {
+  const reportIssueToggle = document.querySelector(".settings-report-issue");
+  const reportIssueSettingsReturn = document.querySelector(".settings-report-issue-return");
+  const submissionSuccessContinue = document.querySelector(".report-continue");
+
+  [reportIssueToggle, reportIssueSettingsReturn, submissionSuccessContinue].forEach(e => {
+    e.addEventListener("click", () => {
+      document.body.classList.toggle("show-report-issue");
+      const eventLabel = document.body.classList.contains("show-report-issue") ? "opened-report-issue" : "closed-report-issue";
+      if (document.body.classList.contains("show-report-issue")) {
+        sendRelayEvent("Panel", "click", eventLabel);
+      }
+    });
+  });
+  const reportForm = document.querySelector('.report-issue-content');
+  reportForm.addEventListener('submit', handleReportIssueFormSubmission);
+  reportURL();
+  showReportInputOtherTextField();
+  showSuccessReportSubmission();
+}
+
+async function handleReportIssueFormSubmission(event) {
+  event.preventDefault();
+  const data = new FormData(event.target);
+  const formJSON = Object.fromEntries(data.entries());
+  
+  Object.keys(formJSON).forEach(function(value) {
+    // Switch "on" to true
+    if (formJSON[value] === "on") {
+      formJSON[value] = true;
+    }
+    // Remove from report if empty string
+    if (formJSON[value] === "") {
+      delete formJSON[value];
+    }
+  });
+
+  await browser.runtime.sendMessage({
+    method: "postReportWebcompatIssue",
+    description: formJSON
+  });
+}
+
+function showSuccessReportSubmission() {
+  const reportIssueSubmitBtn = document.querySelector(".report-issue-submit-btn");
+  const reportSuccess = document.querySelector(".report-success");
+  const reportContent = document.querySelector(".report-issue-content");
+  reportIssueSubmitBtn.addEventListener("click", () => {
+    reportSuccess.classList.remove("is-hidden");
+    reportContent.classList.add("is-hidden");
+  });
+}
+
+async function reportURL() {
+  // Add Site URL placeholder
+  const currentPage = await getCurrentPage();
+  const url = new URL(currentPage.url);
+  const inputFieldUrl = document.querySelector('input[name="issue_on_domain"]');
+  inputFieldUrl.value = url.origin;
```

Comment:
This ensures that the site is autofilled and is in the `https://` format, or backend cannot accept it.

Reply:
Do you mean that this ensures the protocol is part of the URL (which is OK), or that `http://` is rejected and only `https://` is accepted (which would be weirder)?

Reply:
Sorry - both `http://` and `https://` are accepted.